monkey/agent_plugins/credentials_collectors/chrome/src/windows_credentials_database_processor.py

```diff
@@ -1,16 +1,130 @@
-from typing import Collection
+import logging
+import shutil
+import sqlite3
+import tempfile
+from collections.abc import Callable, Collection, Iterator
+from pathlib import Path
+from typing import Optional, TypeAlias
 
-from common.credentials import Credentials
+from Cryptodome.Cipher import AES
+
+from common.credentials import Credentials, EmailAddress, Password, Username
 from common.types import Event
+from infection_monkey.utils.threading import interruptible_iter
 
 from .browser_credentials_database_path import BrowserCredentialsDatabasePath
+from .windows_decryption import win32crypt_unprotect_data
+
+logger = logging.getLogger(__name__)
+
+
+DATABASE_QUERY = "SELECT username_value, password_value FROM logins"
+
+ExtractedCredentialPair: TypeAlias = tuple[str, bytes]
+
+
+def get_logins_from_database(database_path: Path) -> Iterator[ExtractedCredentialPair]:
+    with tempfile.NamedTemporaryFile() as temporary_database_file:
+        temporary_database_path = Path(tempfile.gettempdir()) / temporary_database_file.name
```

Comment:
I'm confused. This should be totally unnecessary. As written, I would expect that temporary_database_path == temporary_database_file.

Reply:
The output of TemporaryFile() (and NamedTemporaryFile()) is a file-like object, not a path

Reply:
### Windows
```
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from tempfile import NamedTemporaryFile
>>> with NamedTemporaryFile() as f:
...     print(f.name)
...
C:\Users\user\AppData\Local\Temp\tmp2wbt816l
>>>
```
### Linux
```
Python 3.11.4 (main, Jun  7 2023, 12:45:48) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from tempfile import NamedTemporaryFile
>>> with NamedTemporaryFile() as f:
...     print(f.name)
... 
/home/user/tmp/tmpfje98t90
>>>
```


The code here works because when the pathlib encounters an absolute path component (i.e. something starting with `/` or `C:\`), it throws out everything that came before.

![image](https://github.com/guardicore/monkey/assets/19957806/43755c95-c862-41a8-a185-fb12c636489d)
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath

Reply:
Hmm, but according to the documentation, we shouldn't de able to use it to open the file again on windows:

```
That name can be retrieved from the name attribute of the returned file-like object. Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows)
```